backend/tournesol/views/criteria_correlations.py

```diff
@@ -0,0 +1,83 @@
+
+import numpy as np
+from django.http import JsonResponse
+from drf_spectacular.utils import OpenApiResponse, OpenApiTypes, extend_schema
+from rest_framework import serializers
+from rest_framework.generics import GenericAPIView
+from scipy.stats import linregress
+
+from tournesol.models.poll import Poll
+from tournesol.models.ratings import ContributorRating
+
+
+def compute_correlation(scores_1: dict, scores_2: dict) -> int:
+    """
+    inputs are dictionaries with entities as keys and scores as values
+    for a given user and a given rating criteria
+    """
+
+    # At least two datapoints are necessary to compute correlations
+    if sum(entity in scores_1 for entity in scores_2) > 1:
+        h = np.corrcoef(
+            [scores_1[e] for e in scores_1 if e in scores_2],
+            [scores_2[e] for e in scores_1 if e in scores_2],
+        )
+        return None if np.isnan(h[0][1]) else h[0][1] 
+    else:
+        return None
+
+
+def compute_slope(scores_1: dict, scores_2: dict) -> int:
+    """ 
+    inputs are dictionaries with entities as keys and scores as values
+    for a given user and a given rating criteria
+    """
+
+    # At least two datapoints are necessary to compute slopes
+    if sum(entity in scores_1 for entity in scores_2) > 1:
+        return linregress(
+            [scores_1[e] for e in scores_1 if e in scores_2],
+            [scores_2[e] for e in scores_1 if e in scores_2],
+        ).slope
+    else:
+        return None
+
+
+class ContributorCriteriaCorrelationsSerializer(serializers.Serializer):
+    criterias = serializers.ListField(child=serializers.CharField(), default=list)
+    correlations = serializers.ListField(child=serializers.ListField(child=serializers.FloatField(), default=list), default=list)
+    slopes = serializers.ListField(child=serializers.ListField(child=serializers.FloatField(), default=list), default=list)
+    
+
+class ContributorCriteriaCorrelationsView(GenericAPIView):
+    
+    @extend_schema(
+        responses={
+            200: ContributorCriteriaCorrelationsSerializer
+        }
+    )
+    def get(self, request, poll_name):
+        poll = Poll.objects.get(name=poll_name)
```

Comment:
Using an non-existing poll raises an error 500.

As we use the `poll_name` URL path parameter in a lot of views, we have created a `PollScopedViewMixin` class that you can use to handle everything for you.

It will retrieve the poll from the URL, and will make it available in the view and the serializer context. It also takes care of returning a proper 404 when the requested poll name doesn't exist.

I suggest:
1. to make the view inherit from `PollScopedViewMixin` and use `self.poll_from_url` 
2. add a test ensuring 404 is returned instead of 500

You can check `ComparisonDetailApi` to see a usage of `PollScopedViewMixin` and `ComparisonApiTestCase.test_authenticated_cant_read_non_existing_poll` for an example of test.

Reply:
Thanks! Done in a0435a02